vector/src/Data/Vector/Generic.hs

```diff
@@ -2615,3 +2615,4 @@ dataCast f = gcast1 f
 
 -- $setup
 -- >>> :set -XFlexibleContexts
+-- >>> default (Int)
```

Comment:
Same thing here, this `default (Int)` is probably not needed.
```suggestion
-- >>> :set -Wno-type-defaults
```

Reply:
See https://github.com/haskell/vector/pull/398#discussion_r685596220.